pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -1198,3 +1202,91 @@ func TestFairEvaluationForNodes(t *testing.T) {
 		}
 	}
 }
+
+func TestPreferNominatedNodeFilterCallCounts(t *testing.T) {
+	tests := []struct {
+		name                  string
+		feature               featuregate.Feature
+		pod                   *v1.Pod
+		nodeReturnCodeMap     map[string]framework.Code
+		expectedCount         int32
+		expectedPatchRequests int
+	}{
+		{
+			name:              "Enable the feature, pod has the nominated node set, filter is called only once",
+			feature:           features.PreferNominatedNode,
+			pod:               st.MakePod().Name("p_with_nominated_node").UID("p").Priority(highPriority).NominatedNodeName("node1").Obj(),
+			nodeReturnCodeMap: map[string]framework.Code{},
```

Comment:
can we leave it `nil` as the third test?

Reply:
Good catch! removed.